cli/cppcheckexecutor.cpp

```diff
@@ -67,6 +67,18 @@
 #include <windows.h>
 #endif
 
+static const std::set<std::string> criticalErrors{
```

Comment:
I do not agree about adding `missingInclude`. Just fixing all `missingInclude` will not make the analysis "complete". We will always have false negatives. Our parser is far from perfect. I will never promise that we have "complete" analysis.

More include paths are not always better. It can have a significant performance penalty. System headers should not be included in general. And I have the feeling that if the header code is very complex it can be better to leave it out.

Reply:
For the error ids I listed, the analysis is stopped. The checkers are not executed at all.

Reply:
About adding a `fatal` severity. In theory it sounds good to me. It's just that it's one more thing that will break plugins and scripts.

Reply:
> I do not agree about adding missingInclude. We will always have false negatives. Our parser is far from perfect. I will never promise that we have "complete" analysis.

In theory if we would manage to use the clang parser I guess we could in theory promise that certain semantic checks was "complete" someday. However, I am not sure if we can use the "clang import" approach then. Maybe we would need to write a clang/gcc plugin instead.

Reply:
Just off the top of my head (it's based on something I considered for another improvement).

- add `Severity::fatal` and change the errors in question
- add `Settings::fatalErrors` (default: `false`)
- and command-line option `--fatal-errors`
- add support for `fatal` to `--enable`/`--disable`
- if that option is not set treat them simply as `error`
- always enable that option for the GUI analysis

This way you can opt-in/opt-out and it fits in the existing frameworks.